Client/Application/AppDelegate.swift

```diff
@@ -226,9 +226,34 @@ class AppDelegate: UIResponder, UIApplicationDelegate, UIViewControllerRestorati
         _ = ChronTabsUserResearch()
         // Leanplum setup
 
-        if let profile = self.profile, LeanPlumClient.shouldEnable(profile: profile) {
-            LeanPlumClient.shared.setup(profile: profile)
-            LeanPlumClient.shared.set(enabled: true)
+        if let profile = self.profile {
+            // Leanplum setup
+            if LeanPlumClient.shouldEnable(profile: profile) {
+                LeanPlumClient.shared.setup(profile: profile)
+                LeanPlumClient.shared.set(enabled: true)
+            }
+            
+            let persistedCurrentVersion = InstallType.persistedCurrentVersion()
+            let introScreen = profile.prefs.intForKey(PrefsKeys.IntroSeen)
+            // upgrade install - Intro screen shown & persisted current version does not match
+            if introScreen != nil && persistedCurrentVersion != AppInfo.appVersion {
+                InstallType.set(type: .upgrade)
+                InstallType.updateCurrentVersion(version: AppInfo.appVersion)
+            }
+            
+            // We need to check if the app is a clean install to use for
+            // preventing the What's New URL from appearing.
+            if introScreen == nil {
+                // fresh install - Intro screen not yet shown
+                InstallType.set(type: .fresh)
+                InstallType.updateCurrentVersion(version: AppInfo.appVersion)
+                // Profile and leanplum setup
+                profile.prefs.setString(AppInfo.appVersion, forKey: LatestAppVersionProfileKey)
+                LeanPlumClient.shared.track(event: .firstRun)
+            } else if profile.prefs.boolForKey(PrefsKeys.KeySecondRun) == nil {
+                profile.prefs.setBool(true, forKey: PrefsKeys.KeySecondRun)
```

Comment:
Where is this pref used?

Reply:
@dnarcese Its used to track 2nd run in Leanplum see the line below it

Reply:
What do we use it for in leanplum?

Reply:
I am not really sure its purpose, maybe something that marketing uses but this was added long time ago so didn't wanna break anything when I moved it from Leanplum integration file.